src/plugins/intel_gpu/src/graph/assign.cpp

```diff
@@ -19,6 +21,23 @@ layout assign_inst::calc_output_layout(const assign_node& node, kernel_impl_para
     return impl_param.typed_desc<assign>()->output_layout;
 }
 
+template<typename ShapeType>
+std::vector<layout> assign_inst::calc_output_layouts(const assign_node& /*node*/, const kernel_impl_params& impl_param) {
```

Comment:
Can't it be implemented via `return forward_input0_shape<ShapeType>(impl_param);` ?

Reply:
Yes, Since `ReadValue` and `Assign` just propagate the input shape as it is, I updated it to use `forward_input0_shape` in header